### PR TITLE
Classroom redux refactor

### PR DESCRIPTION
**Pull request by kylerostler**
## Description

- This code connects all components located inside src/pages/classroom to the Redux store in order to continue to standardize the codebase and move everything towards Redux
- Classroom/Index.js is now flushed out with Redux and consumes data from the backend in order to display courses based on real data in the database
- Upon navigating to FeedbackBadgePage it will now render badges based on the data in the backend as well in so that instructors can actually see the badges a student has 
- Each feedback badge updates live from the backend, adding or removing badges makes requests to the backend and the page then rerenders to display changes immediately, these changes were made so that an instructor can have an easier time using the interface and so that the badges are stored in the database instead of using dummy data
- All the feedbackBadges have conditional rendering checking if a student has a particular badge to decide which version of the feedbackBadge to display that way an instructor can see what badges they can add and which ones they can remove
- Constructed classroomActions and classroomReducer in order to dispatch actions through redux inside the classroom components so that they are Reduxized and to continue to further implement Redux into all components

TLDR; Instructors can now access a particular course and then select a student from that course in order to assign or remove feedback badges to that particular student and that information will be stored in the database.

CO-Author: Will Phillips https://github.com/WRWPhillips
Fixes # (issue)

## Loom Video
Kyler Ostler--
https://www.loom.com/share/b19b96147d574e6b80d760d1e9634163

Will Phillips--
https://www.loom.com/share/dabdc8536d0b42f0821922596d3a5a90

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by bezzantsam — APPROVED** [review]
Great job connecting the badge features to the backend. Excellent use of Redux to consume the data in real time.

**Review by AndrewGary — APPROVED** [review]
Wow massive amount of great work in this PR, not only does the site look good but the redux code is clean and easy to understand.

**Review by jimjamesjimathy — APPROVED** [review]
Excellent work connecting the front-end to the back-end and getting the badges to update immediately, and also the color-changing upon clicking the button is a great indicator that the page has updated. Great work.

**Comment by WRWPhillips** [comment]
Thanks Paul!
